src/networkWrapper/wrappers/socketio.js

```diff
@@ -12,11 +22,75 @@ function SocketIO(host, port, ssl) {
    * @param {int} reconnectionDelay
    */
   this.connect = function (autoReconnect, reconnectionDelay) {
+    var self = this;
+
     this.socket = window.io((this.ssl ? 'https://' : 'http://') + this.host + ':' + this.port, {
       reconnection: autoReconnect,
       reconnectionDelay: reconnectionDelay,
       forceNew: true
     });
+
+    this.socket.on('connect', function() {
+      if (self.wasConnected) {
```

Comment:
Nope, this is different cases:
- reconnection on `connect` event is used when proxy close connection
- reconnection on `reconnect` event is used when socket hang out and come back again via socket io reconnection logic